src/Files.App/Actions/Start/PinToStartAction.cs

```diff
@@ -37,8 +36,12 @@ public async Task ExecuteAsync()
 			{
 				foreach (ListedItem listedItem in context.ShellPage.SlimContentPage.SelectedItems)
 				{
-					var folder = await StorageService.GetFolderAsync(listedItem.ItemPath);
-					await StartMenuService.PinAsync(folder, listedItem.Name);
+					IStorable storable = listedItem.IsFolder switch
+					{
+						true => await StorageService.GetFolderAsync(listedItem.ItemPath),
+						_ => await StorageService.GetFileAsync(listedItem.ItemPath)
+					};
+					await StartMenuService.PinAsync(storable, listedItem.Name);
```

Comment:
I think this fix should be separated from the others and merged immediately, because crashes on pinning are one of the most frequent crashes.

Reply:
Agreed